server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobStore.java

```diff
@@ -70,14 +71,14 @@ private synchronized Path buildAndCreate(BlobPath path) throws IOException {
     }
 
     private Path buildPath(BlobPath path) {
-        String[] paths = path.toArray();
-        if (paths.length == 0) {
+        List<String> paths = path.parts();
+        if (paths.isEmpty()) {
             return path();
         }
-        Path blobPath = this.path.resolve(paths[0]);
-        if (paths.length > 1) {
-            for (int i = 1; i < paths.length; i++) {
-                blobPath = blobPath.resolve(paths[i]);
+        Path blobPath = this.path.resolve(paths.get(0));
+        if (paths.size() > 1) {
```

Comment:
This `if` isn't necessary, we check the condition in the `for` loop anyway.

Reply:
++